src/uri/serializer/uriserializer.rs

```diff
@@ -41,22 +41,22 @@ pub trait UriSerializer<T> {
     /// Serializes a `UUri` into a specific serialization format.
     ///
     /// # Arguments
-    /// * `uri` - The `UUri` object to be serialized into the format `T`.
+    /// * `uri` - The uri object to be serialized into the format `T`.
     ///
     /// # Returns
-    /// Returns a `Result<T, SerializationError>` representing the serialized `UUri` in the specified format.
+    /// Returns the serialized uri in the specified format.
     ///
     /// # Errors
     ///
     /// Returns a `SerializationError` if the serialization process fails. This may be due to reasons such as incompatible data
-    /// in the `UUri` that cannot be represented in the desired format, or errors that occur during the serialization process.
+    /// in the uri that cannot be represented in the desired format, or errors that occur during the serialization process.
     fn serialize(uri: &UUri) -> Result<T, SerializationError>;
 
     /// Builds a fully resolved `UUri` from the serialized long format and the serialized micro format.
     ///
     /// # Arguments
-    /// * `long_uri` - `UUri` serialized as a string.
-    /// * `micro_uri` - `UUri` serialized as a byte slice.
+    /// * `long_uri` - uri serialized as a string.
+    /// * `micro_uri` - uri serialized as a byte slice.
     ///
     /// # Returns
     /// Returns an `Option<UUri>` object serialized from one of the forms. Returns `None` if the URI
```

Comment:
FMPOV we should change this function to return a Result instead of an Option to make it more explicit that this actually constitutes an erroneous situation.

Reply:
Good point, done